src/Searchable/Aggregator.php

```diff
@@ -168,9 +168,11 @@ public static function makeAllSearchable()
 
             $instance->newQuery()->when($softDeletes, function ($query) {
                 $query->withTrashed();
-            })->orderBy($instance->getKeyName())->get()->map(function ($model) {
-                return static::create($model);
-            })->searchable();
+            })->orderBy($instance->getKeyName())->chunk(config('scout.chunk.searchable', 500), function ($models) {
+                $models->filter->shouldBeSearchable()->map(function ($model) {
```

Comment:
I am wondering why scout itself doesn't perform the validation of `shouldBeSearchable` here? 🧐

Reply:
Actually this was slightly incorrect, however, I just updated the order of operations here to ensure that it hits the right objects shouldBeSearchable

Reply:
@jimbojsb @nunomaduro Sorry to bring up an old discussion but I have just been trying to implement an Aggregator in my application and all the `shouldBeSearchable()` methods never get run on the specified Models.

If I change the code to your original suggestion, everything works just fine like this...

```php
$instance->newQuery()->when($softDeletes, function ($query) {
                $query->withTrashed();
            })->orderBy($instance->getKeyName())->chunk(config('scout.chunk.searchable', 500), function ($models) {
                $models = $models->filter->shouldBeSearchable()->map(function ($model) {
                    return static::create($model);
                });
```

The way it currently works is that it tries to run a `shouldBeSearchable()` on the actual Aggregator class which I don't think is the correct implementation as I presume most Models have their own `shouldBeSearchable` criteria.

I'll submit a PR later this evening  to further clarify the situation.

Or am I missing something?

Thanks

Reply:
This also could be related to #103

Reply:
And just for clarification, the command I was having issue with was `artisan scout:import "App\Search\News"`

Reply:
PR here #121